src/main/scala/cognite/spark/v1/AssetsRelation.scala

```diff
@@ -88,22 +89,22 @@ class AssetsRelation(config: RelationConfig)(val sqlContext: SQLContext)
       } *> IO.unit
   }
 
-  def resolveConflict(existingExternalIds: Seq[String], assets: Seq[Asset]): IO[Unit] = {
+  def resolveConflict(existingExternalIds: Seq[String], assets: Seq[AssetCreate]): IO[Unit] = {
     val (assetsToUpdate, assetsToCreate) = assets.partition(
       p => if (p.externalId.isEmpty) { false } else { existingExternalIds.contains(p.externalId.get) }
     )
 
     val idMap = client.assets
       .retrieveByExternalIds(existingExternalIds)
-      .unsafeRunSync()
-      .map(a => a.externalId -> a.id)
-      .toMap
+      .map(_.map(e => e.externalId -> e.id).toMap)
```

Comment:
We should probably use a for Asset here.

Reply:
I think Emil is going to add updating by external ID in the scala sdk, so then we can just use external ID without needing to do this at all, right?

Reply:
Yep!